internal/exec/stages/mount/mount.go

```diff
@@ -117,10 +118,19 @@ func (s stage) mountFs(fs types.Filesystem) error {
 		return err
 	}
 
-	if err := s.Logger.LogOp(func() error { return syscall.Mount(fs.Device, path, *fs.Format, 0, "") },
-		"mounting %q at %q with type %q", fs.Device, path, *fs.Format,
+	args := translateOptionSliceToString(fs.MountOptions)
+	if err := s.Logger.LogOp(func() error { return syscall.Mount(fs.Device, path, *fs.Format, 0, args) },
+		"mounting %q at %q with type %q and options %q", fs.Device, path, *fs.Format, args,
 	); err != nil {
 		return err
 	}
 	return nil
 }
+
+func translateOptionSliceToString(opts []types.MountOption) string {
```

Comment:
Minor: it would be useful to take `,` as an argument here, and also have some basic tests to cover the empty-slice case and a simple populated-slice too.

Reply:
Will add the separator as a separate argument.

Regarding the slice: I'm no Go expert, but from what I know and see Go's `Join` function will handle both the empty and single slice automatically (i.e. return an empty string / a string without comma). Or do you mean testing the functionality via Blackbox tests?